app/code/Magento/Inventory/Model/Command/AssignSourcesToStock.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\Command;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Inventory\Setup\InstallSchema;
+use Magento\InventoryApi\Api\Command\AssignSourcesToStockInterface;
+use Magento\InventoryApi\Api\Data\SourceStockLinkInterface;
+use Psr\Log\LoggerInterface;
+
+
+/**
+ * @inheritdoc
+ */
+class AssignSourcesToStock implements AssignSourcesToStockInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param ResourceConnection $connection
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ResourceConnection $connection,
+        LoggerInterface $logger
+    ) {
+        $this->connection = $connection;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $sourceIds, $stockId)
+    {
+        if (empty($sourceIds)) {
```

Comment:
we need to check both $stockId and $source$ids
and not just empty check, but that both Stock and Sources already exists


And throw InputException otherwise